handlers/deploy.go

```diff
@@ -110,12 +128,42 @@ func deploy(ctx context.Context, req types.FunctionDeployment, client *container
 
 	log.Printf("Container ID: %s\tTask ID %s:\tTask PID: %d\t\n", container.ID(), task.ID(), task.Pid())
 
-	ip, netErr := getIP("netns0", int(task.Pid()))
+	id := uuid.New().String()
+	netns := fmt.Sprintf(NetNSPathFmt, task.Pid())
+
+	cni, err := gocni.New(gocni.WithPluginConfDir(CNIConfDir),
+		gocni.WithPluginDir([]string{CNIBinDir}))
+
+	if err != nil {
+		return err
+	}
+
+	// Load the cni configuration
+	if err := cni.Load(gocni.WithLoNetwork, gocni.WithDefaultConf); err != nil {
+		log.Fatalf("Failed to load cni configuration: %v", err)
+	}
+
+	labels := map[string]string{}
+
+	result, err := cni.Setup(ctx, id, netns, gocni.WithLabels(labels))
+	if err != nil {
+		log.Fatalf("failed to setup network for namespace %q: %v", id, err)
+	}
+
+	var ips []net.IP
+	for name, ifConfig := range result.Interfaces {
+		log.Printf("Config of interface %s: %v\n",
+			name, ifConfig)
 
-	if netErr != nil {
-		return netErr
+		if ifConfig.Sandbox == netns {
+			for _, ipConfig := range ifConfig.IPConfigs {
+				ips = append(ips, ipConfig.IP)
+			}
+		}
 	}
 
+	ip := &ips[0]
```

Comment:
How do we know that ip[0] is always the expected IP? Can we look at the interface's name?

Reply:
We get the IP from CNI default interface now.